controller/source/gdpr_metadata.hpp

```diff
@@ -4,11 +4,28 @@
 #include <algorithm>
 #include <vector>
 #include <unordered_map>
+#include <chrono>
+#include <iomanip>
+#include <sstream>
 
 namespace controller {
 
 constexpr int num_users = 64;
 constexpr int num_purposes = 64;
+constexpr int metadata_prefix_fields = 8;
+
+enum metadata_fields {
+  usr,
+  encr,
+  pur,
+  obj,
+  org,
+  exp,
+  shr,
+  log,
+  val,
+  max_enum_guard
+};
```

Comment:
Just for me to understand, `metadata_prefix_fields` is set to 8 but we have 10 elements in the enum. Is this intentional?

Reply:
The `metadata_prefix_fields` indicates how many fields are appended as a prefix in a value and is used for parsing these fields.
The enum has the 9th member `val` which actually indicates the actual value index when parsing and spliting the fields and a `max_enum_guard` that can be possibly used to verify that we don't exceed that limit. It was not needed so far, but might be useful.